resources/lib/template.py

```diff
@@ -344,7 +344,10 @@ def getProperties( self, elem, items ):
                     continue
                     
                 # We've matched a property :)
-                properties[ name ] = property.text
+                if property.text is not None and property.text != "":
```

Comment:
you can do really crazy stuff in python. Not too long ago I found out that you can put a conditional into the class defintion:
```class ChildClass(ParentClass 1 if condition else ParentClass2)```
I am not sure yet if should find that elegant or ugly. :)